test/unit/core/detail/pack_algorithm_test.cpp

```diff
@@ -7,13 +7,58 @@
 
 #include <gtest/gtest.h>
 
+#include <sstream>
+
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
-#include <seqan3/core/algorithm/parameter_pack.hpp>
+#include <seqan3/core/detail/pack_algorithm.hpp>
 #include <seqan3/core/type_list/type_list.hpp>
 
 using namespace seqan3;
 
-TEST(parameter_pack, for_each_value)
+//-----------------------------------------------------------------------------
+// all_of
+//-----------------------------------------------------------------------------
+
+struct test_integral_meta
```

Comment:
```suggestion
struct is_integral_fn
```

I'm not sure with `_fn`, but it is our naming scheme for view functors.